File: Java/Csharp/AbstractFactory/AbstractFactory/ProductB2.cs
namespace AbstractFactory
{
    class ProductB2: IProductB
    {
        public string UsefulFunctionB()
        {
            return "The result of the product B2.";
        }

        public string AnotherUsefulFunctionB(IProductA collaborator)
        {
            var result = collaborator.UsefulFunctionA();
            return $"The result of the B2 collaborating with the ({result})";
        }
    }
}

File: PHP/Show/ProductB2.php
<?php
include_once('IProductB.php');

class ProductB2 implements IProductB
{

    public function usefulFunctionB(): string
    {
        return "The result of the product B2.";
    }

    public function anotherUsefulFunctionB(IProductA $collaborator): string
    {
        $result = $collaborator->usefulFunctionA();

        return "The result of the B2 collaborating with the ({$result})";
    }
}